packages/bff/src/services/catalogService.ts

```diff
@@ -570,5 +575,112 @@ export function catalogServiceBuilder(
 
       return toBffCatalogApiEserviceRiskAnalysis(riskAnalysis);
     },
+    createDescriptor: async (
+      eServiceId: string,
+      { headers, logger }: WithLogger<BffAppContext>
+    ): Promise<bffApi.CreatedResource> => {
+      const eService = await catalogProcessClient.getEServiceById({
+        params: { eServiceId },
+        headers,
+      });
+
+      if (eService.descriptors.length === 0) {
+        throw noDescriptorInEservice(eServiceId);
+      }
+
+      const maxVersion = (
+        descriptors: catalogApi.EServiceDescriptor[]
+      ): catalogApi.EServiceDescriptor =>
+        descriptors.reduce(
+          (maxVersion, curr) =>
+            parseInt(curr.version, 10) > parseInt(maxVersion.version, 10)
+              ? curr
+              : maxVersion,
+          descriptors[0]
+        );
```

Comment:
Could we rename `maxVersion` to `retrieveLatestDescriptor`? Also inside the function that name is a bit misleading because it might seem that we are retrieving just the version number and not the entire descriptor.

Reply:
990201fab4621358359b0f22a0f473e7f638f4cd